tests/ui/sprite-frame.test.ts

```diff
@@ -0,0 +1,88 @@
+import { SpriteFrame } from "../../cocos/2d";
+import { Texture2D } from "../../cocos/asset/assets";
+import { Rect, Size, Vec2 } from "../../exports/base";
+
+test('spriteframe.clone', () => {
+    let sp = new SpriteFrame;
+
+    sp.insetTop = 100;
+    expect(sp.insetTop).toStrictEqual(100);
+    sp.insetBottom = 200;
+    expect(sp.insetBottom).toStrictEqual(200);
+    sp.insetLeft = 300;
+    expect(sp.insetLeft).toStrictEqual(300);
+    sp.insetRight = 400;
+    expect(sp.insetRight).toStrictEqual(400);
```

Comment:
Why need to check these functions?

Reply:
'_capInsets' is a protected member and does not have an interface to get. So these functions can be tested indirectly.

Reply:
If you want to test other things, i think you should do in another function. This function just test clone.

Reply:
This test needs to be set. If it is not set it will be the default value, and the default value is no guarantee that it is a true copy. 
Checking that it is set correctly and testing the get function can be merged together. Otherwise there will be duplicate work

Reply:
Yep, they need to be set, but don't have to be test here. Maybe you should write other test cases to test the value is set correctly.

Reply:
Ok